docs-standalone/coding-style-guide.md

```diff
@@ -0,0 +1,276 @@
+- The `/* Copyright header */` and the `#import` section.
+- The `#import` section and the class `@interface` or `@implementation` line (or its associated *forward declarations* or *HeaderDoc comment*).
+- Different `@interface` or `@implementation` sections within the same file.
+
+Do not use two or more empty lines in any other cases.
+
+Use exactly one empty line to separate:
+- The last *forward declaration* and the *`@interface` declaration*.
+- The *`@interface` declaration* and the first *method* or *property declaration*.
+- The last *method* or *property declaration* and the *`@end` keyword*.
+- The *`@implementation` line* and the first *method definition*.
+- The closing bracket of the last *method definition* and the *`@end` keyword*.
+- The *`@param` section* and the *`@return` line* in a *[HeaderDoc](https://developer.apple.com/library/mac/documentation/DeveloperTools/Conceptual/HeaderDoc/intro/intro.html) comment*.
+
+*Header* and *implementation* files must have one, and only one, trailing empty line.
+
+Do not use empty lines to separate:
+- *Forward declarations* from other contiguous *forward declarations*.
+- A *HeaderDoc comment* from its related *class* or *method*.
+- Contiguous *`@param` lines* within the same *HeaderDoc comment*.
+- Last statement in a method definition and its closing bracket.
+
+Note that *forward declarations* should appear before any *class HeaderDoc comment*.
+
+You can optionally use one (and only one) blank like to separate:
+- Groups of related `#import` statements.
+- Groups of related statements in a single method implementation.
+
+##### Header File Example
+
+    /*
+     Copyright (c) 2015, Apple Inc. All rights reserved.
+     ...
+     */
+
+
+    #import <Foundation/Foundation.h>
+    #import <ResearchKit/ORKDefines.h>
+
+
+    NS_ASSUME_NONNULL_BEGIN
+
+    @protocol ORKTask;
+    @class ORKStep;
+
+    /**
+     HeaderDoc class comment.
+     */
+    ORK_CLASS_AVAILABLE
+    @interface ORKMyClass : NSObject <NSSecureCoding, NSCopying>
+
+    /**
+     HeaderDoc method comment.
+     
+     @param parameterA   The first parameter.
+     @param parameterB   The second parameter.
+     
+     @return A new MyClass.
+     */
+    - (instancetype)initWithParameterA:(NSString *)parameterA parameterB:(NSString *)parameterB NS_DESIGNATED_INITIALIZER;
+
+    /**
+     HeaderDoc property comment.
+     */
+    @property (nonatomic, copy, readonly) NSString *aProperty;
+
+    @end
+
+
+    /**
+     HeaderDoc class comment.
+     */
+    ORK_CLASS_AVAILABLE
+    @interface ORKMyOtherClass : ORKMyClass
+
+    @end 
+
+    NS_ASSUME_NONNULL_END
+
+
+### 2. Code Style
+
+#### 2.1. Variable Declarations
+
+Declare one variable per line even if they have the same type. In general it's a good idea to initialize variables with a reasonable value.
+
+    // DO
+    int variable1 = -1;
+    int variable2 = -1;
+    int *variablePointer = NULL;
+    id object = nil;
+    NSString *string = nil;
+
```

Comment:
Using ARC, strong, weak, and autoreleasing stack variables are now implicitly initialized with nil.

https://developer.apple.com/library/ios/releasenotes/ObjectiveC/RN-TransitioningToARC/Introduction/Introduction.html#//apple_ref/doc/uid/TP40011226-CH1-SW5

Can we drop this nil assignment requirement? Since it is not required.